packages/cdktf-cli/lib/error-reporting.ts

```diff
@@ -83,6 +91,35 @@ export async function initializErrorReporting(askForConsent = false) {
     autoSessionTracking: true,
     dsn: process.env.SENTRY_DSN,
     release: `cdktf-cli-${DISPLAY_VERSION}`,
+    async beforeSend(event, hint) {
+      if (!hint) {
+        return event;
+      }
+
+      // The promise character is not documented, but it happens
+      const originalException:
+        | Promise<Error>
+        | Error
+        | string
+        | null
+        | undefined = hint.originalException;
+      let error: Error | string | null | undefined;
+      if (isPromise(originalException)) {
+        (originalException as unknown as Promise<Error>).catch(
+          (e) => (error = e)
+        );
+        await Promise.allSettled([originalException]);
+      } else {
+        error = originalException;
+      }
+
+      const errorMessage = error?.toString() || "";
+      if (errorMessage.includes("Usage Error")) {
```

Comment:
Is there a reason you didn't use `IsErrorType`? (I used that elsewhere for a similar of check)
https://github.com/hashicorp/terraform-cdk/blob/b0c6a9547a7480c7e90dce76e1981c93e677a4fd/packages/cdktf-cli/lib/errors.ts#L19